packages/http-server/src/serializers.ts

```diff
@@ -18,4 +20,11 @@ export default [
     },
     serializer: (data: unknown) => (typeof data === 'string' ? data : xmlSerializer.parse(data)),
   },
+  {
+    regex: /text\/plain/,
+    serializer: (data: unknown) => {
+      if (typeof data === 'string') return data;
+      throw ProblemJsonError.fromTemplate(NOT_ACCEPTABLE, 'Cannot serialise complex objects as text/plain');
```

Comment:
It’s late here and I can’t think of which error it should be, but I don’t think 406 is appropriate. The server is failing to create a representation for something defined in the document, and that puts it more towards a 5xx error than something the user did wrong, no?

Reply:
The client is requesting content that is not possible, but that doesn't mean it is the clients fault. 

A 406 means "you cannot accept this, you should accept something that we support". This API (according to the OpenAPI document) supports text/plain, but in trying to request this thing the client is being told it is not supported.

Surly the fact that the client cannot possibly change anything on their end to make this situation work means it is not a client error?

Reply:
Absolutely, 500 is a good 🤷‍♂ which I think is the closest status code to this rather odd situation.